infra/modules/service/main.tf

```diff
@@ -113,13 +126,20 @@ resource "aws_ecs_task_definition" "app" {
           "awslogs-region"        = data.aws_region.current.name,
           "awslogs-stream-prefix" = local.log_stream_prefix
         }
-      }
-      mountPoints    = []
+      },
+      mountPoints    = [for e in local.ephemeral_write_volumes : e.mount_point]
       systemControls = []
       volumesFrom    = []
     }
   ])
 
+  dynamic "volume" {
+    for_each = [for e in local.ephemeral_write_volumes : e.volume]
+    content {
+      name = volume.value["name"]
```

Comment:
```suggestion
    for_each = [for volume_config in local.ephemeral_write_volume_configs : volume_config.volume]
    content {
      name = volume_config.value["name"]
```

Reply:
For [dynamic blocks](https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks) the `for_each` var doesn't determine the iterator name, but can make it a little more compact with:

```
  dynamic "volume" {
    for_each = local.ephemeral_write_volume_configs[*].volume
    content {
      name = volume.value["name"]
    }
  }
```

As I don't think this is any clearer:
```
  dynamic "volume" {
    for_each = [for config in local.ephemeral_write_volume_configs : config.volume]
    iterator = "config"
    content {
      name = config.value["name"]
    }
  }
```
And we don't currently use `iterator` in our other dynamic blocks.

Reply:
Ah ok yeah the first code block looks good